server.py

```diff
@@ -7,16 +7,21 @@ def main():
     serv.listen(5)
     while True:
         conn, addr = serv.accept()
-        print 'Client from address: {} conn: {}'.format(addr, conn)
+        print 'Client from address: {} '.format(addr)
         data = "Start"
         while data != "":
-            data = conn.recv(4096)
+            try:
+                data = conn.recv(4096)
+            except:
```

Comment:
Catching a broad exception is bad practice, you want to specify the type of exception you want, in this case, it's probably socket.error, that way, you can also print some of the exception details to the user